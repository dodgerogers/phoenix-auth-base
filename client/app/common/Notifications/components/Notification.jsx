import React from "react";
import PropTypes from "prop-types";
import ImmutablePropTypes from "react-immutable-proptypes";
import { connect } from "react-redux";
import functional from "react-functional";
import { Message } from "semantic-ui-react";
import * as NotificationActions from "../actions";

const Notification = (props) => {
  const { notification, onDestroy } = props;
  const message = notification.get("message");
  const level = notification.get("level");
  const levelProps = level ? { [level]: true } : {};

  return (
    <Message onDismiss={onDestroy} {...levelProps}>
      {message}
    </Message>
  );
};

Notification.propTypes = {
  onDestroy: PropTypes.func.isRequired,
  notification: ImmutablePropTypes.mapContains({
    message: PropTypes.string.isRequired,
    level: PropTypes.string,
  }),
};

const opts = {
  componentDidMount: (props) => setTimeout(props.onDestroy, 5000),
};

export { Notification as PureComponent };
export const FunctionalComponent = functional(Notification, opts);
export default connect(null, NotificationActions)(FunctionalComponent);
