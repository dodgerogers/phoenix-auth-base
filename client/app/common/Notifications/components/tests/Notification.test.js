import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import toJson from 'enzyme-to-json';
import Notification, { FunctionalComponent, PureComponent } from '../Notification';


describe('Notification', () => {
  const fullProps = () => ({
    onDestroy: jest.fn(),
    notification: fromJS({
      message: 'Information',
      level: null,
    }),
  });

  describe('FunctionalComponent', () => {
    beforeEach(() => {
      jest.useFakeTimers()
    });

    const functionalComponent = (props) => shallow(
      <FunctionalComponent {...props} />, { lifecycleExperimental: true }
    );

    it('calls props.onDestroy on componentDidMount', () => {
      const props = fullProps();
      const component = functionalComponent(props);

      jest.runAllTimers();

      expect(props.onDestroy).toHaveBeenCalled();
    });
  });

  describe('PureComponent', () => {
    const pureComponent = (props) => shallow(
      <PureComponent {...props} />
    );

    it('matches snapshot with no level given', () => {
      const component = pureComponent(fullProps());

      expect(toJson(component)).toMatchSnapshot();
    });

    it('matches snapshot with success level given', () => {
      const props = fullProps();
      props.notification = props.notification.set('level', 'success');
      const component = pureComponent(props);

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
