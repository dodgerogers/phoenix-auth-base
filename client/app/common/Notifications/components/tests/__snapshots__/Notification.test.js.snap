// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification PureComponent matches snapshot with no level given 1`] = `
<Message
  onDismiss={[Function]}
>
  Information
</Message>
`;

exports[`Notification PureComponent matches snapshot with success level given 1`] = `
<Message
  onDismiss={[Function]}
  success={true}
>
  Information
</Message>
`;
